x/tokenomics/keeper/settle_pending_claims.go

```diff
@@ -265,29 +276,157 @@ func (k Keeper) SettlePendingClaims(ctx sdk.Context) (
 		)
 	}
 
+	if err = k.ExecutePendingResults(ctx, settledResults); err != nil {
+		return settledResults, expiredResults, err
+	}
+
 	// Slash all the suppliers that have been marked for slashing slashingCount times.
-	for supplierOperatorAddress, slashingCount := range supplierToExpiredClaimCount {
-		if err := k.slashSupplierStake(ctx, supplierOperatorAddress, slashingCount); err != nil {
+	// The suppliers are sorted to ensure deterministic iteration.
+	slices.Sort(expiredClaimSupplierOperatorAddresses)
+	for _, supplierOperatorAddress := range expiredClaimSupplierOperatorAddresses {
+		slashingCount := supplierToExpiredClaimCount[supplierOperatorAddress]
+		if err = k.slashSupplierStake(ctx, supplierOperatorAddress, slashingCount); err != nil {
 			logger.Error(fmt.Sprintf("error slashing supplier %s: %s", supplierOperatorAddress, err))
-			return settledResult, expiredResult, err
+			return settledResults, expiredResults, err
 		}
 	}
 
 	logger.Info(fmt.Sprintf(
 		"settled %d and expired %d claims at block height %d",
-		settledResult.NumClaims,
-		expiredResult.NumClaims,
+		settledResults.GetNumClaims(),
+		expiredResults.GetNumClaims(),
 		blockHeight,
 	))
 
-	return settledResult, expiredResult, nil
+	return settledResults, expiredResults, nil
+}
+
```

Comment:
The logic in `Keeper#ExecutePendingSettledResults` and its dependencies is new. Instead of modifying state in the TLMs, we now build up the `SettlementResult` object and then execute all of its state transitions at  the end. `#ExecutePendingSettledResults` (and now `#ExecutePendingExpiredResults`) are the roots of this series of state transitions.